File: tests/expectations/both/struct.c
#include <stdint.h>
#include <stdlib.h>
#include <stdbool.h>

typedef struct Opaque Opaque;

typedef struct Normal {
  int32_t x;
  float y;
} Normal;

typedef struct NormalWithZST {
  int32_t x;
  float y;
} NormalWithZST;

typedef struct TupleRenamed {
  int32_t m0;
  float m1;
} TupleRenamed;

typedef struct TupleNamed {
  int32_t x;
  float y;
} TupleNamed;

void root(Opaque *a, Normal b, NormalWithZST c, TupleRenamed d, TupleNamed e);

File: tests/expectations/struct.zig
const std = @import("std");

const Opaque = opaque {

};

pub const Normal = extern struct {
  x: i32,
  y: f32,
};

pub const NormalWithZST = extern struct {
  x: i32,
  y: f32,
};

pub const TupleRenamed = extern struct {
  m0: i32,
  m1: f32,
};

pub const TupleNamed = extern struct {
  x: i32,
  y: f32,
};

pub extern fn root( a: ?*Opaque,
                   b: Normal,
                   c: NormalWithZST,
                   d: TupleRenamed,
                   e: TupleNamed) anyopaque;
